examples/MomentoApplication/Program.cs

```diff
@@ -6,43 +6,94 @@
-        static readonly uint DEFAULT_TTL_SECONDS = 60;
-
-        async static Task Main(string[] args)
+        if (createError.ErrorCode == MomentoErrorCode.LIMIT_EXCEEDED_ERROR)
+        {
+            Console.WriteLine("Error: cache limit exceeded. We need to talk to support@moentohq.com! Exiting.");
+        } else
         {
-            using SimpleCacheClient client = new SimpleCacheClient(Configurations.Laptop.Latest, MOMENTO_AUTH_TOKEN, DEFAULT_TTL_SECONDS);
-            var createCacheResult = await client.CreateCacheAsync(CACHE_NAME);
+            Console.WriteLine($"Error creating cache: {createError.Message}. Exiting.");
+        }
+        // Any error is considered fatal.
+        Environment.Exit(1);
+    }
+    // If there's already a cache by this name, we don't want to modify it.
+    if (createCacheResponse is CreateCacheResponse.CacheAlreadyExists)
+    {
+        Console.WriteLine($"Error: strangely, a cache with the name {CACHE_NAME} already exists. Exiting.");
+        Environment.Exit(1);
+    }
 
-            Console.WriteLine("Listing caches:");
-            String token = null;
-            do
-            {
-                ListCachesResponse resp = await client.ListCachesAsync(token);
-                if (resp is ListCachesResponse.Success successResult)
-                {
-                    foreach (CacheInfo cacheInfo in successResult.Caches)
-                    {
-                        Console.WriteLine(cacheInfo.Name);
-                    }
-                    token = successResult.NextPageToken;
-                }
-            } while (!String.IsNullOrEmpty(token));
-            Console.WriteLine($"\nSetting key: {KEY} with value: {VALUE}");
-            await client.SetAsync(CACHE_NAME, KEY, VALUE);
-            Console.WriteLine($"\nGet value for  key: {KEY}");
-            CacheGetResponse getResponse = await client.GetAsync(CACHE_NAME, KEY);
-            if (getResponse is CacheGetResponse.Hit hitResponse)
+    Console.WriteLine("\nListing caches:");
+    String token = null;
+    do
+    {
+        ListCachesResponse response = await client.ListCachesAsync(token);
+        if (response is ListCachesResponse.Success successResponse)
+        {
+            foreach (CacheInfo cacheInfo in successResponse.Caches)
             {
-                Console.WriteLine($"\nLookedup value: {hitResponse.String()}, Stored value: {VALUE}");
+                Console.WriteLine($"- {cacheInfo.Name}");
             }
+            token = successResponse.NextPageToken;
+        } else if (response is ListCachesResponse.Error listError)
+        {
+            // We do not consider this a fatal error, so we just report it.
+            Console.WriteLine($"Error listing caches: {listError.Message}");
+            break;
         }
+    } while (!String.IsNullOrEmpty(token));
+
+    Console.WriteLine($"\nSetting key: {KEY} with value: {VALUE}");
+    var setResponse = await client.SetAsync(CACHE_NAME, KEY, VALUE);
```

Comment:
If we want to use `setResponse` as a var name, we might want to use in other places as well such as `listResponse` so that it's more coherent.

Reply:
Addressed in a07686e.